### PR TITLE
Fix #17: Migrate source to Kotlin

### LINKED ISSUE #17: Use kotlin as plugin language (instead of Groovy)
Question to plugin owner: probably it is better to use Kotlin language for plugin next development?

Why:
* It is first-class citizen of Gradle for now
* We can have strong-typed language for:
** Configuration (with IntelliSence at the IntelliJ Idea)
** Plugin development
* It is a bit faster than groovy (because of compilation)

### PR DESCRIPTION
**Pull request by mikejhill**
Took a stab at migrating source to Kotlin. Let me know if this seems helpful. I tried to keep this to refactoring and minimize functional changes.

As users of the plugin, we would like to take advantage of some Gradle features that are easier to implement using Kotlin, e.g. #39.

### CODE REVIEW

**Comment by deepy** [comment]
From a cursory glance on my phone this looks great!
I'm on vacation right now but I'll have a proper look on Monday

File: src/main/kotlin/com/moowork/gradle/node/exec/ExecRunner.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            // Take care of Windows environments that may contain "Path" OR "PATH" - both existing
+            // possibly (but not in parallel as of now)
+            val envPathName = if (execEnvironment["Path"] != null) "Path" else "PATH"
+            execEnvironment[envPathName] = path + File.pathSeparator + execEnvironment[envPathName]
+        }
+        return execEnvironment
+    }
+
+    // Configurable; Groovy support
+    fun setExecOverrides(execOverrides: Closure<ExecSpec>) {
```
Comment (deepy) [inline src/main/kotlin/com/moowork/gradle/node/exec/ExecRunner.kt]: note to self: I think we might be able to replace `Closure` with `Action`

File: build.gradle [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+	apply from: "$rootDir/gradle/buildscript.gradle", to: buildscript
 }
 
-group = 'com.github.node-gradle'
+plugins {
+	id('idea')
+	id('groovy')
+	id('maven-publish')
+	id('org.gradle.kotlin.kotlin-dsl') version "1.3.1"
+	id('com.cinnober.gradle.semver-git') version "2.4.0"
```
Comment (deepy) [inline build.gradle]: note to self: see if we have to move the configuration for semver-git when we move to `plugins {}`

**Comment by deepy** [comment]
I've looked through it and it seems great, just gotta run some manual tests before merging!

**Comment by bsautel** [comment]
Thanks @mikejhill for this great pull request! And thanks @deepy for reviewing it!

**Comment by deepy** [comment]
Thank you for all the work @mikejhill!

I'll be able to do a thorough test on the 20th when I got access to a project that's really good at triggering corner-cases
